src/components/Table/GridTable.tsx

```diff
@@ -356,16 +377,50 @@ export function GridTable<R extends Kinded, X extends Only<GridTableXss, X> = an
     const filteredRows = filterRows(api, columns, maybeSorted, filter);
     visitRows(filteredRows, 0, true);
 
-    return [headerRows, visibleDataRows, totalsRows, expandableHeaderRows, filteredRowIds];
-  }, [as, api, filter, maybeSorted, columns, style, rowStyles, sortOn, columnSizes, collapsedIds, getCount]);
+    // Check for any selected rows that are not displayed in the table because they don't match the current filter, or are no longer part of the `rows` prop.
+    // We persist these selected rows and hoist them to the top of the table.
+    if (sortedUnmatchedSelections.length) {
+      const hsrTuples: RowTuple<R>[] = sortedUnmatchedSelections.map((row, idx) => [
```

Comment:
Thought: Kinda wonder if `hsrTuples` could just be inlined; took me a little while to read this section of the code, so looking for small simplifications.

Reply:
It appears if I inline it, then I need to also cast it `as RowTuple<R>[]`, which feels a bit dangerous, but probably fine. Otherwise the inferred type is `(GridDataRow<R> | Element)[]`, and causes a TS error.

Reply:
ooOOOooo I could probably use `satisfies`. I'll try that.